### PR TITLE
Add vscode configurations to make development easier

### PR DESCRIPTION
**Pull request by Tabrizian**
Now you can start development on Python backend just by cloning the folder and using dev container.

Some tasks are added for configuration and build.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+				"ms-vscode.cpptools-extension-pack",
+				"ms-vscode.cmake-tools"
+			]
+		}
+	},
+
+	"runArgs": [ "--cap-add=SYS_PTRACE", "--security-opt", "seccomp=unconfined", "--gpus=all", "--shm-size=2g", "--ulimit", "stack=67108864"],
+	"mounts": [
+		"source=${localEnv:HOME}/.ssh,target=/home/triton-server/.ssh,type=bind,consistency=cached",
+		"source=${localEnv:HOME}/scratch,target=/scratch,type=bind,consistency=cached",
```
Comment (fpetrini15) [inline .devcontainer/devcontainer.json:22]: I did not have this when I tried to open the folder in a container and it caused an error. Is this a reasonable default mount?

Comment (Tabrizian) [reply .devcontainer/devcontainer.json]: Good point. Removed.

File: .vscode/tasks.json [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            "command": "sudo",
+            "args": [
+                "cp",
+                "-r",
+                "${workspaceFolder}/build/install/backends/python/*",
+                "/opt/tritonserver/backends/python"
+            ]
+        },
+        {
+            "label": "Build Python",
```
Comment (fpetrini15) [inline .vscode/tasks.json:73]:
```suggestion
            "label": "Build Python Backend",
```
Nit: To me, this would be a little more clear. Thoughts?

**Review by fpetrini15 — APPROVED** [review]
Tested with a new build and was able to build the python backend successfully. Optional nit.

Beyond that, LGTM! 🚀

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+FROM nvcr.io/nvidia/tritonserver:24.03-py3
+
+ARG USERNAME=triton-server
+
+RUN apt-get update \
+    && apt-get install -y sudo
+
+# RUN pip3 install transformers torch
```
Comment (krishung5) [inline .devcontainer/Dockerfile:34]: Quick question - I see that this line was uncommented at the beginning but now commented. Is this something we would want to install by default, or we just wanted to leave it as is and developers can decide to have these installed or not.

File: .vscode/tasks.json (hunk 2 above)
Comment (krishung5) [reply .vscode/tasks.json:73]: I'd vote for "Build Python Backend" too.

**Review by krishung5 — APPROVED** [review]
Confirmed that I can also build the backend locally without issues. Thanks Iman, nice improvement!

File: .devcontainer/Dockerfile (hunk 3 above)
Comment (Tabrizian) [reply .devcontainer/Dockerfile:34]: Good catch. Fixed.

File: README.md [hunk 4]
```diff
@@ -1810,6 +1811,16 @@ def initialize(self, args):
     # Should print {'custom_key': {'string_value': 'custom_value'}}
 ```
 
+# Development with VSCode
+
+The repository includes a `.devcontainer` folder that contains a `Dockerfile`
+and `devcontainer.json` file to help you develop the Python backend
+using Visual Studio Code.
+
+In order to build the backend, you can the "Build Python Backend" task in the
```
Comment (fpetrini15) [inline README.md:1821]:
```suggestion
In order to build the backend, you can execute the "Build Python Backend" task in the
```

**Comment by fpetrini15** [comment]
Rebuilt with the new changes (works!) but noticed this time around that we get a warning that "TRITON_ENABLE_ENSEMBLE" is unused.

**Review by fpetrini15 — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (Tabrizian) [reply README.md]: good catch fixed.

**Comment by Tabrizian** [comment]
> Rebuilt with the new changes (works!) but noticed this time around that we get a warning that "TRITON_ENABLE_ENSEMBLE" is unused.

I removed the unused cmake variables.

**Review by fpetrini15 — APPROVED** [review]
LGTM!

**Pull request merged**